pkg/cloudprovider/providers/aws/aws.go

```diff
@@ -2671,9 +2677,18 @@ func (a *AWSCloud) getInstancesByIDs(instanceIDs []*string) (map[string]*ec2.Ins
 	return instancesByID, nil
 }
 
-// Fetches instances by node names; returns an error if any cannot be found.
+// Fetches instances by node names; returns an error if any cannot be found. Input must be sorted.
 // This is implemented with a multi value filter on the node names, fetching the desired instances with a single query.
 func (a *AWSCloud) getInstancesByNodeNames(nodeNames []string) ([]*ec2.Instance, error) {
```

Comment:
How about if we rename this `getInstancesByNodeNamesCached`, and add a TODO to rationalize all the caching in 1.4?

I think having an instance cache is a good idea, but I'm not crazy about the idea of having this cached but the single-fetch not cached (inconsistency) and I'm worried about the safety of exposing the whole cached ec2.Instance object (I think it's safe now, but we're setting ourselves up for weird problems later).

I also think that you should just drop the requirement that nodeNames be sorted; I think we just create a sets.String instead (for reasons that will become even clearer in my next few notes!)